### ISSUE TITLE
Gzip not working in quarkus resteasy-reactive.

### ISSUE THREAD

**Issue by itzsuman** [quarkusio/quarkus#34379]
okay, in that case, a `ContainerResponseFilter` can help, when you use resteasy-reactive. 
For resteasy classic (the quarkus-resteasy extensions), I am not sure if this gets invoked, since files are served from a vert.x handler before the resteasy classic handler has any chance to run, afaik.

```
@Provider
@PreMatching
public class GZIPResponseFilter implements ContainerResponseFilter {
    @Override
    public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext) throws IOException {
        if (requestContext.getUriInfo().getPath().contains("WebGL.framework.js.gz")) {
            responseContext.getHeaders().putSingle("Content-Encoding", "gzip");
        }
    }
}
```

_Originally posted by @Postremus in https://github.com/quarkusio/quarkus/discussions/23162#discussioncomment-2040953_

**Comment by itzsuman** [comment]
Hi, I want to compressed or gzip my rest-api response ,  added ContainerResponseFilter  with Content-Encoding: gzip but I am getting "incorrect header check " error . Please help.
Also added  quarkus.http.enable-compression in my configuration property. Please help.

**Comment by itzsuman** [comment]
> okay, in that case, a `ContainerResponseFilter` can help, when you use resteasy-reactive. For resteasy classic (the quarkus-resteasy extensions), I am not sure if this gets invoked, since files are served from a vert.x handler before the resteasy classic handler has any chance to run, afaik.
> 
> ```
> @Provider
> @PreMatching
> public class GZIPResponseFilter implements ContainerResponseFilter {
>     @Override
>     public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext) throws IOException {
>         if (requestContext.getUriInfo().getPath().contains("WebGL.framework.js.gz")) {
>             responseContext.getHeaders().putSingle("Content-Encoding", "gzip");
>         }
>     }
> }
> ```
> 
> _Originally posted by @Postremus in [#23162 (reply in thread)](https://github.com/quarkusio/quarkus/discussions/23162#discussioncomment-2040953)_


 @Postremus  I found this post of yours but with  Content-Encoding: gzip getting error. Could you please help.

**Comment by Postremus** [comment]
@itzsuman Please try the steps mentioned in https://quarkus.io/guides/resteasy-reactive#http-compression.

However, I must say that section was difficult to find with google, maybe we should fix that? I.e. it does not mention gzip anywhere on that page.

**Comment by Postremus** [comment]
@itzsuman 

I got it compression working.
TL;DR: you need to do a full restart of your dev mode before enabling compression takes effect. I.e. STRG+C in your terminal, and start quarkus again in dev mode


FTR. My application.properties:
```
quarkus.http.enable-compression=true
```

Resource:
```
    @Path("/hello")
public class GreetingResource {

    @GET
    @Compressed
    public String hello() {
        char[] chars = new char[100000];
        Arrays.fill(chars, 'a');
        return new String(chars);
    }
```

**Comment by itzsuman** [comment]
yes complete restart only but I am using 2.15.0.Final and quarkus-resteasy-reactive. Could you please share your version or pom.

**Comment by Postremus** [comment]
@itzsuman 
my full sample project:
[gzip-reactive.zip](https://github.com/quarkusio/quarkus/files/11901914/gzip-reactive.zip)

**Comment by itzsuman** [comment]
@Postremus Thank you. 
yes its working . 
If I add any of the below mentioned dependency , gzip is not working .
quarkus-resteasy-reactive-jackson
quarkus-rest-client-reactive-jackson
quarkus-amazon-lambda-rest

**Comment by Postremus** [comment]
@itzsuman 
you are right, once quarkus-amazon-lambda-rest is added to the sample project, gzip is not working.

updated sample:
[gzip-reactive.zip](https://github.com/quarkusio/quarkus/files/11904691/gzip-reactive.zip)

To reproduce:
mvn clean compile quarkus:dev
open a browser
open devtools / network tab
visit http://localhost:8080/hello/2
in devtools, you can see that no compression used for the response

/cc @geoand Could you take a look please?

**Comment by geoand** [comment]
I will take a look soon

**Comment by itzsuman** [comment]
@geoand @Postremus even if I add any jackson dependency then also gzip not working.

**Comment by itzsuman** [comment]
> @Postremus Thank you. yes its working . If I add any of the below mentioned dependency then gzip is not working . quarkus-resteasy-reactive-jackson quarkus-rest-client-reactive-jackson quarkus-amazon-lambda-rest

@geoand Could you please help me on this.

**Comment by itzsuman** [comment]
> I will take a look soon


@geoand Did you get a chance to look into this issue. Plz help.

**Comment by geoand** [comment]
I have not

**Comment by geoand** [comment]
So this looke like a duplicate of https://github.com/quarkusio/quarkus/issues/34472, correct?

**Comment by itzsuman** [comment]
@geoand yes similar but reported #34472 as a bug and also in amazon-lambda level.
